src/Dscp/Witness/Launcher/Runner.hs

```diff
@@ -7,18 +6,17 @@ import Universum
 
 import Control.Monad.Component (runComponentM)
 
-import Dscp.Launcher.Resource (AllocResource (..))
+import Dscp.Resource.Class (AllocResource (..))
 import Dscp.Witness.Launcher.Mode (WitnessContext (..), WitnessRealMode)
 import Dscp.Witness.Launcher.Params (WitnessParams (..))
 import Dscp.Witness.Launcher.Resource (WitnessResources (..))
 
+-- TODO Maybe this function should be "-> IO WitnessContext" and other
+-- non-resource context parts can be allocated here.
```

Comment:
What are non-resource context parts which require IO to be allocated? Can you provide an example?

Reply:
Maybe, but it can be changed later easily if needed, we would hardly use this function outside of the module :woman_shrugging: